test/parallel/test-stringbytes-external-at-max.js

```diff
@@ -1,22 +1,27 @@
 'use strict';
+// Flags: --expose-gc
 
 require('../common');
 const assert = require('assert');
 
+assert(typeof gc === 'function', 'Run this test with --expose-gc');
+
 // v8 fails silently if string length > v8::String::kMaxLength
 // v8::String::kMaxLength defined in v8.h
 const kStringMaxLength = process.binding('buffer').kStringMaxLength;
 
+var buf;
 try {
-  new Buffer(kStringMaxLength * 3);
+  buf = new Buffer(kStringMaxLength);
+  // Try to allocate memory first then force gc so future allocations succeed.
+  new Buffer(2 * kStringMaxLength);
+  gc();
 } catch(e) {
-  assert.equal(e.message, 'Invalid array buffer length');
+  if (e.message != 'Invalid array buffer length') throw (e);
```

Comment:
I'd add a comment explaining you don't use assert in order to preserve the original exception.